File: mt_items.py
from org.bukkit import Material
from org.bukkit.inventory import ItemStack

# Helper function to create an ItemStack with custom model data
def create_custom_item(material, custom_model_data):
    item = ItemStack(material)
    item_meta = item.getItemMeta()
    item_meta.setCustomModelData(custom_model_data)
    item.setItemMeta(item_meta)
    return item

# Create custom items
human_teleporter = create_custom_item(Material.PAPER, 8008)
cold_fusion_reactor = create_custom_item(Material.PAPER, 8009)
infuser = create_custom_item(Material.PAPER, 8010)
pollution_generator = create_custom_item(Material.PAPER, 8011)
sifter = create_custom_item(Material.PAPER, 8012)
crafter = create_custom_item(Material.PAPER, 8013)
pump = create_custom_item(Material.PAPER, 8014)
moonlight_panel = create_custom_item(Material.PAPER, 8015)
solar_panel = create_custom_item(Material.PAPER, 8016)
auto_collector = create_custom_item(Material.PAPER, 8017)
elevator = create_custom_item(Material.PAPER, 8018)
item_teleporter = create_custom_item(Material.PAPER, 8019)
auto_timer = create_custom_item(Material.PAPER, 8020)
auto_shearer = create_custom_item(Material.PAPER, 8021)
auto_milker = create_custom_item(Material.PAPER, 8022)
auto_breeder = create_custom_item(Material.PAPER, 8023)
deep_drill = create_custom_item(Material.PAPER, 8024)
bow_turret = create_custom_item(Material.PAPER, 8025)
auto_plucker = create_custom_item(Material.PAPER, 8026)
auto_planter = create_custom_item(Material.PAPER, 8027)
auto_logger = create_custom_item(Material.PAPER, 8028)
auto_harvester = create_custom_item(Material.PAPER, 8029)
auto_miner = create_custom_item(Material.PAPER, 8030)
incinerator = create_custom_item(Material.PAPER, 8031)
auto_butcher = create_custom_item(Material.PAPER, 8032)
power_reciever = create_custom_item(Material.PAPER, 8033)
advanced_bow_turret = create_custom_item(Material.PAPER, 8034)
laser_turret = create_custom_item(Material.PAPER, 8035)
biter_detector = create_custom_item(Material.PAPER, 8036)
redstone_emitter = create_custom_item(Material.PAPER, 8037)
heat_exchanger = create_custom_item(Material.PAPER, 8038)
nuclear_reactor = create_custom_item(Material.PAPER, 8039)
power_meter = create_custom_item(Material.PAPER, 8040)
power_transmitter = create_custom_item(Material.PAPER, 8041)
item_forge = create_custom_item(Material.PAPER, 8042)
battery = create_custom_item(Material.PAPER, 8043)
battery_monitor = create_custom_item(Material.PAPER, 8044)
power_pylon = create_custom_item(Material.PAPER, 8045)
power_pylon_mk2 = create_custom_item(Material.PAPER, 8046)
power_substation = create_custom_item(Material.PAPER, 8047)
centrifuge = create_custom_item(Material.PAPER, 8048)
production_monitor = create_custom_item(Material.PAPER, 8049)
combustion_generator = create_custom_item(Material.PAPER, 8050)
oil_refinery = create_custom_item(Material.PAPER, 8051)
boiler = create_custom_item(Material.PAPER, 8052)
steam_engine = create_custom_item(Material.PAPER, 8053)
petroleum_engine = create_custom_item(Material.PAPER, 8054)
basic_assembler = create_custom_item(Material.PAPER, 8055)
advanced_assembler = create_custom_item(Material.PAPER, 8056)
research_lab = create_custom_item(Material.PAPER, 8057)
electric_furnace = create_custom_item(Material.PAPER, 8058)
overdriver = create_custom_item(Material.PAPER, 8059)
train_station = create_custom_item(Material.PAPER, 8060)
auto_placer = create_custom_item(Material.PAPER, 8061)
auto_breaker = create_custom_item(Material.PAPER, 8062)
geothermal_generator = create_custom_item(Material.PAPER, 8063)
wind_turbine = create_custom_item(Material.PAPER, 8064)
tidal_generator = create_custom_item(Material.PAPER, 8065)
heavy_steam_generator = create_custom_item(Material.PAPER, 8066)
roboport = create_custom_item(Material.PAPER, 8067)
auto_cauldron = create_custom_item(Material.PAPER, 8068)
mine_layer = create_custom_item(Material.PAPER, 8069)
barrel_pump = create_custom_item(Material.PAPER, 8070)
flood_light = create_custom_item(Material.PAPER, 8071)
mover = create_custom_item(Material.PAPER, 8072)

File: src/main/java/me/darksoul/mTBlockModels/MTItems.java
package me.darksoul.mTBlockModels;

import org.bukkit.Material;
import org.bukkit.inventory.ItemStack;
import org.bukkit.inventory.meta.ItemMeta;

public class MTItems {

    public static final ItemStack HUMAN_TELEPORTER = createCustomItem(Material.PAPER, 8008);
    public static final ItemStack COLD_FUSION_REACTOR = createCustomItem(Material.PAPER, 8009);
    public static final ItemStack INFUSER = createCustomItem(Material.PAPER, 8010);
    public static final ItemStack POLLUTION_GENERATOR = createCustomItem(Material.PAPER, 8011);
    public static final ItemStack SIFTER = createCustomItem(Material.PAPER, 8012);
    public static final ItemStack CRAFTER = createCustomItem(Material.PAPER, 8013);
    public static final ItemStack PUMP = createCustomItem(Material.PAPER, 8014);
    public static final ItemStack MOONLIGHT_PANEL = createCustomItem(Material.PAPER, 8015);
    public static final ItemStack SOLAR_PANEL = createCustomItem(Material.PAPER, 8016);
    public static final ItemStack AUTO_COLLECTOR = createCustomItem(Material.PAPER, 8017);
    public static final ItemStack ELEVATOR = createCustomItem(Material.PAPER, 8018);
    public static final ItemStack ITEM_TELEPORTER = createCustomItem(Material.PAPER, 8019);
    public static final ItemStack AUTO_TIMER = createCustomItem(Material.PAPER, 8020);
    public static final ItemStack AUTO_SHEARER = createCustomItem(Material.PAPER, 8021);
    public static final ItemStack AUTO_MILKER = createCustomItem(Material.PAPER, 8022);
    public static final ItemStack AUTO_BREEDER = createCustomItem(Material.PAPER, 8023);
    public static final ItemStack DEEP_DRILL = createCustomItem(Material.PAPER, 8024);
    public static final ItemStack BOW_TURRET = createCustomItem(Material.PAPER, 8025);
    public static final ItemStack AUTO_PLUCKER = createCustomItem(Material.PAPER, 8026);
    public static final ItemStack AUTO_PLANTER = createCustomItem(Material.PAPER, 8027);
    public static final ItemStack AUTO_LOGGER = createCustomItem(Material.PAPER, 8028);
    public static final ItemStack AUTO_HARVESTER = createCustomItem(Material.PAPER, 8029);
    public static final ItemStack AUTO_MINER = createCustomItem(Material.PAPER, 8030);
    public static final ItemStack INCINERATOR = createCustomItem(Material.PAPER, 8031);
    public static final ItemStack AUTO_BUTCHER = createCustomItem(Material.PAPER, 8032);
    public static final ItemStack POWER_RECEIVER = createCustomItem(Material.PAPER, 8033);
    public static final ItemStack ADVANCED_BOW_TURRET = createCustomItem(Material.PAPER, 8034);
    public static final ItemStack LASER_TURRET = createCustomItem(Material.PAPER, 8035);
    public static final ItemStack BITER_DETECTOR = createCustomItem(Material.PAPER, 8036);
    public static final ItemStack REDSTONE_EMITTER = createCustomItem(Material.PAPER, 8037);
    public static final ItemStack HEAT_EXCHANGER = createCustomItem(Material.PAPER, 8038);
    public static final ItemStack NUCLEAR_REACTOR = createCustomItem(Material.PAPER, 8039);
    public static final ItemStack POWER_METER = createCustomItem(Material.PAPER, 8040);
    public static final ItemStack POWER_TRANSMITTER = createCustomItem(Material.PAPER, 8041);
    public static final ItemStack ITEM_FORGE = createCustomItem(Material.PAPER, 8042);
    public static final ItemStack BATTERY = createCustomItem(Material.PAPER, 8043);
    public static final ItemStack BATTERY_MONITOR = createCustomItem(Material.PAPER, 8044);
    public static final ItemStack POWER_PYLON = createCustomItem(Material.PAPER, 8045);
    public static final ItemStack POWER_PYLON_MK2 = createCustomItem(Material.PAPER, 8046);
    public static final ItemStack POWER_SUBSTATION = createCustomItem(Material.PAPER, 8047);
    public static final ItemStack CENTRIFUGE = createCustomItem(Material.PAPER, 8048);
    public static final ItemStack PRODUCTION_MONITOR = createCustomItem(Material.PAPER, 8049);
    public static final ItemStack COMBUSTION_GENERATOR = createCustomItem(Material.PAPER, 8050);
    public static final ItemStack OIL_REFINERY = createCustomItem(Material.PAPER, 8051);
    public static final ItemStack BOILER = createCustomItem(Material.PAPER, 8052);
    public static final ItemStack STEAM_ENGINE = createCustomItem(Material.PAPER, 8053);
    public static final ItemStack PETROLEUM_ENGINE = createCustomItem(Material.PAPER, 8054);
    public static final ItemStack BASIC_ASSEMBLER = createCustomItem(Material.PAPER, 8055);
    public static final ItemStack ADVANCED_ASSEMBLER = createCustomItem(Material.PAPER, 8056);
    public static final ItemStack RESEARCH_LAB = createCustomItem(Material.PAPER, 8057);
    public static final ItemStack ELECTRIC_FURNACE = createCustomItem(Material.PAPER, 8058);
    public static final ItemStack OVERDRIVER = createCustomItem(Material.PAPER, 8059);
    public static final ItemStack TRAIN_STATION = createCustomItem(Material.PAPER, 8060);
    public static final ItemStack AUTO_PLACER = createCustomItem(Material.PAPER, 8061);
    public static final ItemStack AUTO_BREAKER = createCustomItem(Material.PAPER, 8062);
    public static final ItemStack GEOTHERMAL_GENERATOR = createCustomItem(Material.PAPER, 8063);
    public static final ItemStack WIND_TURBINE = createCustomItem(Material.PAPER, 8064);
    public static final ItemStack TIDAL_GENERATOR = createCustomItem(Material.PAPER, 8065);
    public static final ItemStack HEAVY_STEAM_GENERATOR = createCustomItem(Material.PAPER, 8066);
    public static final ItemStack ROBOPORT = createCustomItem(Material.PAPER, 8067);
    public static final ItemStack AUTO_CAULDRON = createCustomItem(Material.PAPER, 8068);
    public static final ItemStack MINE_LAYER = createCustomItem(Material.PAPER, 8069);
    public static final ItemStack BARREL_PUMP = createCustomItem(Material.PAPER, 8070);
    public static final ItemStack FLOOD_LIGHT = createCustomItem(Material.PAPER, 8071);
    public static final ItemStack MOVER = createCustomItem(Material.PAPER, 8072);
    public static final ItemStack CRANK_GENERATOR = createCustomItem(Material.PAPER, 8073);
    public static final ItemStack LIGHTNING_GENERATOR = createCustomItem(Material.PAPER, 8074);
    public static final ItemStack MOB_BLOCKER = createCustomItem(Material.PAPER, 8075);
    public static final ItemStack CRUSHER = createCustomItem(Material.PAPER, 8076);
    public static final ItemStack ROCKET_SILO = createCustomItem(Material.PAPER, 8077);
    public static final ItemStack CHUNK_LOADER = createCustomItem(Material.PAPER, 8078);
    public static final ItemStack ARTILLERY_DEVICE = createCustomItem(Material.PAPER, 8079);

    private static ItemStack createCustomItem(Material material, int customModelData) {
        ItemStack item = new ItemStack(material);
        ItemMeta itemMeta = item.getItemMeta();
        if (itemMeta != null) {
            itemMeta.setCustomModelData(customModelData);
            item.setItemMeta(itemMeta);
        }
        return item;
    }
}
